include/lk/snappy_boot_v1.h

```diff
@@ -0,0 +1,117 @@
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef _BOOTLOADER_SNAP_BOOT_V1_H
+#define _BOOTLOADER_SNAP_BOOT_V1_H
+
+#define SNAP_BOOTSELECT_VERSION   0x00010001
+#define SNAP_BOOTSELECT_SIGNATURE ('S' | ('B' << 8) | ('s' << 16) | ('e' << 24))
+#define SNAP_NAME_MAX_LEN (256)
+#define HASH_LENGTH (32)
+#define SNAP_MODE_LENGTH (8)
+#define SNAP_MODE_TRY "try"
+#define SNAP_MODE_TRYING "trying"
+#define FACTORY_RESET "factory-reset"
+
+/* partition label where boot select structure is stored */
+#define SNAP_BOOTSELECT_PARTITION "snapbootsel"
+
+/* number of available bootimg partitions, min 2 */
+#define SNAP_BOOTIMG_PART_NUM 2
+
+/* snappy bootselect partition format structure */
+typedef struct SNAP_BOOT_SELECTION {
+    /* Contains value BOOTSELECT_SIGNATURE defined above */
+    uint32_t signature;
+    /* snappy boot select version */
+    uint32_t version;
+
+    /* snap_mode, one of: 'empty', "try", "trying" */
+    char snap_mode[SNAP_MODE_LENGTH];
+    /* current core snap revision */
+    char snap_core[SNAP_NAME_MAX_LEN];
+    /* try core snap revision */
+    char snap_try_core[SNAP_NAME_MAX_LEN];
+    /* current kernel snap revision */
+    char snap_kernel[SNAP_NAME_MAX_LEN];
+    /* current kernel snap revision */
+    char snap_try_kernel[SNAP_NAME_MAX_LEN];
+
+    /* GADGET assets: current gadget assets revision */
+    char snap_gadget[SNAP_NAME_MAX_LEN];
+    /* GADGET assets: try gadget assets revision */
+    char snap_try_gadget [SNAP_NAME_MAX_LEN];
+
+    /**
+     Reboot reason
+     Optional parameter to signal bootloader alternative reboot reasons
+     e.g. recovery/factory-reset/boot asset update
+    */
+    char reboot_reason[SNAP_NAME_MAX_LEN];
+
+	/**
+      Matrix for mapping of boot img partion to installed kernel snap revision
+      At image build time:
+        - snap prepare populates:
+             - fills matrix first column with bootimage part names based on
+               gadget.yaml file where we will support multiple occurrences of the role: bootimg
+             - fills boot_part_num with number of actually available boot partitions
+        - snapd:
+             - when new kernel snap is installed, snap updates mapping in matrix so
+               bootloader can pick correct kernel snap to use for boot
+             - snap_mode, snap_try_kernel, snap_try_core behaves same way as with u-boot
+             - boot partition labels are never modified by snapd at run time
+        - bootloader:
+             - Finds boot partition to use based on info in matrix and snap_kernel / snap_try_kernel
+             - bootloaer does not alter matrix, only alters snap_mode
+
+        [ <bootimg 1 part label> ] [ <currently installed kernel snap revison> ]
+        [ <bootimg 2 part label> ] [ <currently installed kernel snap revision> ]
+    */
```

Comment:
the comment here doesn't match the go code one anymore, they need syncing

Reply:
yep, I already synched those with next update